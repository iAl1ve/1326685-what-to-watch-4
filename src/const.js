export const UrlPage = {
  MAIN: `main`,
  MOVIE_PAGE: `movie-page`,
};

export const VIDEO_SIZE = 280;

export const PLAYBACK_DELAY = 1000;

export const MAX_SHOW_MORE_FILMS = 8;
export const MAX_COUNT_FILMS = 4;

export const GENRE_DEFAULT = `All genres`;

export const Tabs = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};

export const MONTH_NAMES = [
  `January`,
  `February`,
  `March`,
  `April`,
  `May`,
  `June`,
  `July`,
  `August`,
  `September`,
  `October`,
  `November`,
  `December`,
];
