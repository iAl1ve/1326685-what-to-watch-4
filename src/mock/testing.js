export const ListMovies = [
  {
    id: 1,
    title: `The Grand Budapest Hotel`,
    genre: `Comedies`,
    year: 1996,
    src: `img/the-grand-budapest-hotel-poster.jpg`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: 8.9,
    ratingCount: 256,
    movieDescription: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
    Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    movieDirector: `Wes Andreson`,
    movieStarring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    runTime: 122,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    isFavorite: false,
    backgroundColor: `#B41D8F`,
    previewImage: `img/the-grand-budapest-hotel-poster.jpg`,
  }, {
    id: 2,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    src: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `Drama`,
    year: 2013,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: 9.1,
    ratingCount: 146,
    movieDescription: `Fantastic Beasts: The Crimes of Grindelwald movie description text. Fantastic Beasts: The Crimes of Grindelwald movie description text. Fantastic Beasts: The Crimes of Grindelwald movie description text. `,
    movieDirector: `Wes Andreson`,
    movieStarring: [`Bill Murray`, `Willem Dafoe and other`],
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    runTime: 131,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    isFavorite: false,
    backgroundColor: `#B41D81`,
    previewImage: `img/the-grand-budapest-hotel-poster.jpg`,
  }, {
    id: 3,
    title: `Bohemian Rhapsody`,
    src: `img/bohemian-rhapsody.jpg`,
    genre: `Comedies`,
    year: 2015,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: 4.1,
    ratingCount: 195,
    movieDescription: `Bohemian Rhapsody movie description text. Bohemian Rhapsody movie description text. Bohemian Rhapsody movie description text. `,
    movieDirector: `Bill Murray`,
    movieStarring: [`Edward Norton`, `Jude Law`, `Willem Dafoe`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    runTime: 103,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    isFavorite: false,
    backgroundColor: `#B31D8F`,
    previewImage: `img/the-grand-budapest-hotel-poster.jpg`,
  }, {
    id: 4,
    title: `Macbeth`,
    src: `img/macbeth.jpg`,
    genre: `Drama`,
    year: 2014,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: 9.3,
    ratingCount: 119,
    movieDescription: `Macbeth movie description text. Macbeth movie description text. Macbeth movie description text. Macbeth movie description text. `,
    movieDirector: `Macbeth`,
    movieStarring: [`Macbeth`, `Willem Dafoe`],
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    runTime: 212,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    isFavorite: false,
    backgroundColor: `#B40D8F`,
    previewImage: `img/the-grand-budapest-hotel-poster.jpg`,
  }, {
    id: 5,
    title: `Aviator`,
    src: `img/aviator.jpg`,
    genre: `Documentary`,
    year: 2013,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: 9.2,
    ratingCount: 249,
    movieDescription: `Aviator movie description text. Aviator movie description text. Aviator movie description text. Aviator movie description text.`,
    movieDirector: `Aviator`,
    movieStarring: [`Bill Murray`, `Aviator`, `Willem Dafoe`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    runTime: 165,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    isFavorite: false,
    backgroundColor: `#A31D8F`,
    previewImage: `img/the-grand-budapest-hotel-poster.jpg`,
  }, {
    id: 6,
    title: `Moonrise Kingdom`,
    src: `img/moonrise-kingdom.jpg`,
    genre: `Romance`,
    year: 2011,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: 9.1,
    ratingCount: 232,
    movieDescription: `Moonrise Kingdom movie description text. Moonrise Kingdom movie description text. Moonrise Kingdom movie description text. Moonrise Kingdom movie description text.`,
    movieDirector: `Moonrise Kingdom`,
    movieStarring: [`Moonrise Kingdom`, `Jude Law`],
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    runTime: 138,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    isFavorite: false,
    backgroundColor: `#C51D8F`,
    previewImage: `img/the-grand-budapest-hotel-poster.jpg`,
  }, {
    id: 7,
    title: `Midnight Special`,
    src: `img/midnight-special.jpg`,
    genre: `Crime`,
    year: 2016,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: 8.4,
    ratingCount: 212,
    movieDescription: `Midnight Special movie description text. Midnight Special movie description text. Midnight Special movie description text. Midnight Special movie description text.`,
    movieDirector: `Midnight Special`,
    movieStarring: [`Midnight Special`, `Jude Law`, `Willem Dafoe`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    runTime: 154,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    isFavorite: false,
    backgroundColor: `#D31D8F`,
    previewImage: `img/the-grand-budapest-hotel-poster.jpg`,
  }, {
    id: 8,
    title: `War of the Worlds`,
    src: `img/war-of-the-worlds.jpg`,
    genre: `Drama`,
    year: 2013,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: 8.9,
    ratingCount: 201,
    movieDescription: `War of the Worlds movie description text. War of the Worlds movie description text. War of the Worlds movie description text. War of the Worlds movie description text. War of the Worlds movie description text. `,
    movieDirector: `War of the Worlds`,
    movieStarring: [`War of the Worlds`],
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    runTime: 141,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    isFavorite: false,
    backgroundColor: `#B40A8F`,
    previewImage: `img/the-grand-budapest-hotel-poster.jpg`,
  }, {
    id: 9,
    title: `Shutter Island`,
    src: `img/shutter-island.jpg`,
    genre: `Horror`,
    year: 2017,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    ratingScore: 7.9,
    ratingCount: 175,
    movieDescription: `Shutter Island movie description text. Shutter Island movie description text. Shutter Island movie description text. Shutter Island movie description text. `,
    movieDirector: `Shutter Island`,
    movieStarring: [`Bill Murray`, `Shutter Island`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    runTime: 109,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    isFavorite: false,
    backgroundColor: `#A61D8F`,
    previewImage: `img/the-grand-budapest-hotel-poster.jpg`,
  }
];

export const ListReviews = [
  {
    id: 1,
    idUser: 2,
    text: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    author: `Kate Muir`,
    date: `2016-12-24`,
    rating: 7.6,
  }, {
    id: 2,
    idUser: 2,
    text: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
    author: `Matthew Lickona`,
    date: `2016-12-20`,
    rating: 8.3,
  }, {
    id: 3,
    idUser: 2,
    text: `Anderson's films are too precious for some, but for those of us willing to lose ourselves in them, they're a delight. "The Grand Budapest Hotel" is no different, except that he has added a hint of gravitas to the mix, improving the recipe.`,
    author: `Bill Goodykoontz`,
    date: `2015-11-18`,
    rating: 7.3,
  },
  {
    id: 4,
    idUser: 3,
    text: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
    author: `Paula Fleri-Soler`,
    date: `2016-12-20`,
    rating: 7.8,
  },
  {
    id: 5,
    idUser: 1,
    text: `When one of Gustave's lovers dies mysteriously. It is certainly a magical and childlike way of storytelling.`,
    author: `Amanda Soler`,
    date: `2016-12-20`,
    rating: 9.6,
  },
  {
    id: 6,
    idUser: 3,
    text: `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.`,
    author: `Amanda Greever`,
    date: `2015-11-18`,
    rating: 8.9,
  },
  {
    id: 7,
    idUser: 4,
    text: `Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    author: `Bill Muir`,
    date: `2016-08-14`,
    rating: 8.4,
  },
  {
    id: 8,
    idUser: 5,
    text: `Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs.`,
    author: `Wes  Dafoe`,
    date: `2015-11-21`,
    rating: 5.6,
  },
  {
    id: 9,
    idUser: 2,
    text: `Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    author: `Kate Norton`,
    date: `2015-03-08`,
    rating: 2,
  },
  {
    id: 10,
    idUser: 1,
    text: `Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    author: `Andreson Fleri`,
    date: `2016-05-09`,
    rating: 10,
  },
];

export const listGenres = [`All genres`, `Comedies`, `Drama`, `Documentary`, `Romance`, `Crime`, `Horror`];

export const notAdaptedListReviews = [
  {
    id: 1,
    user: {
      id: 2,
      name: `Kate Muir`
    },
    comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    date: `2016-12-24`,
    rating: 7.6,
  }
];

export const adaptedUserInfo = {
  id: 1,
  email: `Oliver.conner@gmail.com`,
  name: `Oliver.conner`,
  avatarUrl: `img/1.png`
};
