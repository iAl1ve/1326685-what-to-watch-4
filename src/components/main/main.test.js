import React from "react";
import renderer from "react-test-renderer";
import Main from "./main.jsx";
import {ListMovies, listGenres} from "../../mock/testing.js";
import {GENRE_DEFAULT, MAX_SHOW_MORE_FILMS} from "../../const.js";

describe(`Test Render Main`, () => {
  it(`Render Main`, () => {
    const tree = renderer
      .create(<Main
        movie = {ListMovies[0]}
        listMovies = {ListMovies}
        listGenres = {listGenres}
        currentGenre = {GENRE_DEFAULT}
        countShowMovies = {MAX_SHOW_MORE_FILMS}
        onTitleButtonClick = {() => {}}
        onGenreItemClick = {() => {}}
        onShowMoreClick = {() => {}}
        onPlayerExitClick = {() => {}}
      />, {
        createNodeMock: () => {
          return {};
        }
      }).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
