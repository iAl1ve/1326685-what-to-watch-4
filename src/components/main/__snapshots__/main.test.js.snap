// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Render Main Render Main 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="The Grand Budapest Hotel"
        src="img/bg-the-grand-budapest-hotel.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="User avatar"
            height="63"
            src="img/avatar.jpg"
            width="63"
          />
        </div>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="The Grand Budapest Hotel"
            height="327"
            src="img/the-grand-budapest-hotel-poster.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            The Grand Budapest Hotel
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Comedies
            </span>
            <span
              className="movie-card__year"
            >
              1996
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Comedies
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Drama
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Documentary
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Romance
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Crime
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Horror
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <img
              alt="The Grand Budapest Hotel"
              height="175"
              src="img/the-grand-budapest-hotel-poster.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
              onClick={[Function]}
            >
              The Grand Budapest Hotel
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <img
              alt="Fantastic Beasts: The Crimes of Grindelwald"
              height="175"
              src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
              onClick={[Function]}
            >
              Fantastic Beasts: The Crimes of Grindelwald
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <img
              alt="Bohemian Rhapsody"
              height="175"
              src="img/bohemian-rhapsody.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
              onClick={[Function]}
            >
              Bohemian Rhapsody
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <img
              alt="Macbeth"
              height="175"
              src="img/macbeth.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
              onClick={[Function]}
            >
              Macbeth
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <img
              alt="Aviator"
              height="175"
              src="img/aviator.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
              onClick={[Function]}
            >
              Aviator
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <img
              alt="Moonrise Kingdom"
              height="175"
              src="img/moonrise-kingdom.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
              onClick={[Function]}
            >
              Moonrise Kingdom
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <img
              alt="Midnight Special"
              height="175"
              src="img/midnight-special.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
              onClick={[Function]}
            >
              Midnight Special
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <img
              alt="War of the Worlds"
              height="175"
              src="img/war-of-the-worlds.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
              onClick={[Function]}
            >
              War of the Worlds
            </a>
          </h3>
        </article>
      </div>
      <div
        className="catalog__more"
      >
        <button
          className="catalog__button"
          onClick={[Function]}
          type="button"
        >
          Show more
        </button>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          © 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
